src/CollectionsHandler.js

```diff
@@ -0,0 +1,51 @@
+import { handler } from './utils';
+// TODO: Handle non-list entities
+export function listHandler() {
+  return handler(async (_, path) => {
+    let _path = await path;
+    const list = [];
+    while (`${_path}` !== 'http://www.w3.org/1999/02/22-rdf-syntax-ns#nil') {
```

Comment:
is the stringification necessary?

Reply:
From memory `_path` is an LDflex object - so the options would either be this (which assumes the presence of the `.valueOf` handler) or `(await _path).equals(namedNode('http://www.w3.org/1999/02/22-rdf-syntax-ns#nil'))` which assumes the presence of the `.equals` handler.

Happy to change to the latter if you think it is more appropriate.

Reply:
Right; but this code assumes a specific `toString` handler was well. I think I'd `await` the `.value` and to a string compare, but up to you.